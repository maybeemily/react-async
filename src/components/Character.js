import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  const { name, status, species, image } = character;
  return (
    <section>
      <ul>
        <li><img src={image}/></li>
        <li>Name: {name}</li>
        <li>Status: {status}</li>
        <li>Species: {species}</li>
      </ul>
    </section>
  );

}

Character.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired
};

export default Character;
