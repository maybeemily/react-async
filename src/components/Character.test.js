import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a character', () => {
    const testCharacter = {
      image: '/path',
      name: 'Morty',
      status: 'Alive',
      species: 'Human'
    };
    const wrapper = shallow(<Character character = {testCharacter}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
