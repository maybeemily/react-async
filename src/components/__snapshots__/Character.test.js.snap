// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<section>
  <ul>
    <li>
      <img
        src="/path"
      />
    </li>
    <li>
      Name: 
      Morty
    </li>
    <li>
      Status: 
      Alive
    </li>
    <li>
      Species: 
      Human
    </li>
  </ul>
</section>
`;
